tests/data_snack_dynamic_entity/compound_entity/test_validate.py

```diff
@@ -0,0 +1,90 @@
+import pytest
+from typing import Dict
+
+from src.data_snack_dynamic_entity.validate import (
+    validate_entity_templates,
+    ValidationError,
+)
+from src.data_snack_dynamic_entity.compound_entity.factory import _SCHEMA as COMPOUND_ENTITY_SCHEMA
+
+
+@pytest.fixture
+def compound_entity_templates_missing_required_fields() -> Dict:
+    return {
+        "Registration": {
```

Comment:
I think we should use a format that is more similar to the `CompoundEntity` structure.

```
Registration:
    type: compound
    sources:
        - entity: Car
          fields:
               field: "person_index"
               source_field: "index"
```
And then for normal entities we would have `type: simple` and we could do a factory function that checks the type a right type.

What do you think?